### ISSUE TITLE
[2.5] Rancher deployed Apps incorrectly add io.cattle.field/appId to NetworkPolicy selector specification 

### ISSUE THREAD

**Issue by rawmind0** [rancher/rancher#33659]
Backport https://github.com/rancher/rancher/issues/29272

**Comment by izaac** [comment]
This issue was originally reproduced on `2.5.8` in issue #29272

Rancher version: `2.5-head (07/16//2021)` _0e58768_
On this version the label is not added no the matchers by Rancher.
This match the `2.6 master-head` behavior

```yaml
            matchLabels:
              app.kubernetes.io/component: primary
              app.kubernetes.io/instance: gitea
              app.kubernetes.io/name: postgresql
```

Still need to confirm expected behavior during upgrades.

**Comment by izaac** [comment]
Rancher version: `2.6 master-head (07/19/2021)` _0e58768_

As [mentioned](https://github.com/rancher/rancher/issues/29272#issuecomment-882479426) in the `master-head` issue, after a Rancher upgrade the selectors were kept intact in the already deployed app resources. 

Using `Delete and recreate resources if needed during the upgrade (uses helm --force)` provisions the App without the extra selector as expected in the App's recreated resources in a Rancher build with the fix (`2.5-head`).
